python/lsst/pipe/base/connections.py

```diff
@@ -497,7 +498,8 @@ def adjustQuantum(self, datasetRefMap: NamedKeyDict[DatasetType, typing.Set[Data
         return datasetRefMap
 
 
-def iterConnections(connections: PipelineTaskConnections, connectionType: str) -> typing.Generator:
+def iterConnections(connections: PipelineTaskConnections,
+                    connectionType: Union[str, Iterable[str]]) -> typing.Generator:
```

Comment:
Making the return type `typing.Generator[BaseConnection, None, None]` is a lot more useful than saying `Generator` with no generics.  And I know we might differ on this, but if the user shouldn't care that this is implemented as a generator, consider `Iterator` instance

Reply:
I didn't think to expand this when making the change, but I updated it now